scripts/txt2img.py

```diff
@@ -294,8 +304,11 @@ def main():
                 for n in trange(opt.n_iter, desc="Sampling"):
                     for prompts in tqdm(data, desc="data"):
                         uc = None
-                        if opt.scale != 1.0:
+                        if negative_prompt:
+                            uc = model.get_learned_conditioning(batch_size * [negative_prompt])
```

Comment:
I tried this -- it does not appear to work. E.g., try prompt="dog with a hat" and you will likely see some white hats. Use "white" for negative_prompt. An identical image is generated.

Reply:
> A nearly identical image is generated.

The mechanism is to remove noise from the image to induce "visibility" to the requested image, remove noise from the image to induce "invisibility" to the same image, and generate an image from the difference between them.
Therefore, if you put something that represents the same meaning in both images, the difference between them will not be known, and the image will not be generated.

(I am not good at English, so some parts may be wrong. Sorry.)

Reply:
My mistake -- it does appear to have an effect. The effect is just not what I was expecting given examples I've seen implemented in web UIs.

Does the depth2img mode support this (e.g., scripts/gradio/depth2img.py)? I'm not very familiar with the SD code base but I can see that it seems to use a lower-level pipeline to do its work:

```
       # cond
        cond = {"c_concat": [c_cat], "c_crossattn": [c]}

        # uncond cond
        uc_cross = model.get_unconditional_conditioning(num_samples, "")
        uc_full = {"c_concat": [c_cat], "c_crossattn": [uc_cross]}
        if not do_full_sample:
            # encode (scaled latent)
            z_enc = sampler.stochastic_encode(
                z, torch.tensor([t_enc] * num_samples).to(model.device))
        else:
            z_enc = torch.randn_like(z)
        # decode it
        samples = sampler.decode(z_enc, cond, t_enc, unconditional_guidance_scale=scale,
                                 unconditional_conditioning=uc_full, callback=callback)
```

Where `c` is:

```
c = model.cond_stage_model.encode(batch["txt"])
```

I should probably dig into get_learned_condition() because I assume it must assemble the object that sampler.decode() requires (sampler.sample() will eventually call this).

Reply:
I don't know, I haven't tried it, but I think the depth2img is from SD2.
I'll make a PR for SD2.
Thank you for your valuable input!